From 9dcef5daa7eea3e98ff89b0a1cba0b9113f2062c Mon Sep 17 00:00:00 2001
From: someone5678 <59456192+someone5678@users.noreply.github.com>
Date: Sun, 1 Dec 2024 15:15:35 +0900
Subject: [PATCH] Settings: Expose clipboard auto clear setting [1/3]

Ported to AXP.OS lineage-22.2 (rhodep fork) from GrapheneOS 763b9ec29527.

Change-Id: I31af7c915ae9dd60dd38e3eb5cca928fd41ba6d4
---
 core/java/android/provider/Settings.java      | 12 +++++++++
 .../server/clipboard/ClipboardService.java    | 26 ++++++-------------
 2 files changed, 20 insertions(+), 18 deletions(-)

diff --git a/core/java/android/provider/Settings.java b/core/java/android/provider/Settings.java
index 4f3911ba2..8d6369b35 100644
--- a/core/java/android/provider/Settings.java
+++ b/core/java/android/provider/Settings.java
@@ -12911,6 +12911,18 @@ public final class Settings {
          */
         public static final String HIDE_PRIVATESPACE_ENTRY_POINT = "hide_privatespace_entry_point";
 
+        /**
+         * Whether to enable clipboard auto clear
+         * @hide
+         */
+        public static final String CLIPBOARD_AUTO_CLEAR_ENABLED = "clipboard_auto_clear_enabled";
+
+        /**
+         * Timeout length for clipboard auto clear
+         * @hide
+         */
+        public static final String CLIPBOARD_AUTO_CLEAR_TIMEOUT = "clipboard_auto_clear_timeout";
+
         /**
          * The amount of time in milliseconds before camera is turned off
          * @hide
diff --git a/services/core/java/com/android/server/clipboard/ClipboardService.java b/services/core/java/com/android/server/clipboard/ClipboardService.java
index d95e53e9d..83e817c55 100644
--- a/services/core/java/com/android/server/clipboard/ClipboardService.java
+++ b/services/core/java/com/android/server/clipboard/ClipboardService.java
@@ -118,19 +118,6 @@ public class ClipboardService extends SystemService {
     @VisibleForTesting
     public static final long DEFAULT_CLIPBOARD_TIMEOUT_MILLIS = 3600000;
 
-    /**
-     * Device config property for whether clipboard auto clear is enabled on the device
-     **/
-    public static final String PROPERTY_AUTO_CLEAR_ENABLED =
-            "auto_clear_enabled";
-
-    /**
-     * Device config property for time period in milliseconds after which clipboard is auto
-     * cleared
-     **/
-    public static final String PROPERTY_AUTO_CLEAR_TIMEOUT =
-            "auto_clear_timeout";
-
     // DeviceConfig properties
     private static final String PROPERTY_MAX_CLASSIFICATION_LENGTH = "max_classification_length";
     private static final int DEFAULT_MAX_CLASSIFICATION_LENGTH = 400;
@@ -603,12 +590,16 @@ public class ClipboardService extends SystemService {
             }
         }
 
+        private boolean isAutoClearEnabled() {
+            return Settings.Secure.getInt(getContext().getContentResolver(),
+                Settings.Secure.CLIPBOARD_AUTO_CLEAR_ENABLED, 1) == 1;
+        }
+
         private void scheduleAutoClear(
                 @UserIdInt int userId, int intendingUid, int intendingDeviceId) {
             final long oldIdentity = Binder.clearCallingIdentity();
             try {
-                if (DeviceConfig.getBoolean(DeviceConfig.NAMESPACE_CLIPBOARD,
-                        PROPERTY_AUTO_CLEAR_ENABLED, true)) {
+                if (isAutoClearEnabled()) {
                     Pair<Integer, Integer> userIdDeviceId = new Pair<>(userId, intendingDeviceId);
                     mClipboardClearHandler.removeEqualMessages(ClipboardClearHandler.MSG_CLEAR,
                             userIdDeviceId);
@@ -628,9 +619,8 @@ public class ClipboardService extends SystemService {
         }
 
         private long getTimeoutForAutoClear() {
-            return DeviceConfig.getLong(DeviceConfig.NAMESPACE_CLIPBOARD,
-                    PROPERTY_AUTO_CLEAR_TIMEOUT,
-                    DEFAULT_CLIPBOARD_TIMEOUT_MILLIS);
+            return Settings.Secure.getLong(getContext().getContentResolver(),
+                Settings.Secure.CLIPBOARD_AUTO_CLEAR_TIMEOUT, DEFAULT_CLIPBOARD_TIMEOUT_MILLIS);
         }
 
         @Override
-- 
2.43.0

